### ISSUE TITLE
Cursor 0.33.4 refuses to start on Kubuntu 24.04: "SUID sandbox helper binary was found, but is not configured correctly"

### ISSUE THREAD

**Issue by meladoro** [getcursor/cursor#1465]
1) If you can, please include a screenshot of your problem

Here it is, though it hardly makes sense in this case:

![Screenshot_20240522_171426](https://github.com/getcursor/cursor/assets/58917066/90d32d25-d33a-48d3-a8d2-7f132d9ffd26)

2) Please include the name of your operating system

As reported by `lsb_release -a`:

Distributor ID: Ubuntu
Description: Ubuntu 24.04 LTS
Release: 24.04
Codename: noble

Flavor is Kubuntu, though.

3) If you can, steps to reproduce are super helpful

- Download latest version of Cursor from cursor.sh, cursor-0.33.4x86_64.AppImage
- Run chmod u+x ./cursor-0.33.4x86_64.AppImage
- Run ./cursor-0.33.4x86_64.AppImage
- Cursor refuses to start with following error: `[3940:0523/104843.586137:FATAL:setuid_sandbox_host.cc(158)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /tmp/.mount_cursorl2LAz7/chrome-sandbox is owned by root and has mode 4755.
Trace/breakpoint trap (core dumped)`

I appreciate any assistance. Thank you very much.

m.

**Comment by mostafa20220** [comment]
I have the same issue.
The only thing that worked for me is running the appimage using the flag: `-- no-sandbox` 

I searched a lot for a solution to run it without the no-sandbox flag ... but found no thing useful yet.

I hope someone find a better solution than the no-sandbox flag.

**Comment by Octatedron-Alum** [comment]
Same here.

**Comment by rehypothecation** [comment]
Having this issue too.

**Comment by 0xlidar** [comment]
This happens after Ubuntu 24.04

To resolve this issue, cursor developers need to add an apparmor profile like this https://github.com/mullvad/mullvadvpn-app/pull/6196

**Comment by dushmanta05** [comment]
Same issue with me Ubuntu 24.04

**Comment by helgster77** [comment]
I have the same sandbox issue on Ubuntu 24.04 and if I run with "--no-sandbox" the UI starts up but I get SQL Lite errors and some others. Has anyone solved these problems and can share step by step instructions? Given all the hype about Cursor I'm kind of surprised, Ubuntu is not exactly a niche operating system.

**Comment by dushmanta05** [comment]
@helgster77 I've exactly same issue on my Ubuntu 24.04

**Comment by Imobacho** [comment]
Same here

**Comment by jurjendevries** [comment]
Same with Ubuntu 24.04. I am able to run it a bit with --no-sandbox, but it crashes randomly.

**Comment by Buckler89** [comment]
Same here with ubuntu 24.04

**Comment by nettum** [comment]
I just updated from 22.04 to 24.04 and get the same error.
As far as I can see there are at least a few issues regarding this - this one, #1483 and #1683

**Comment by nettum** [comment]
FWIW, as far as I can see vscode runs with the `--no-sandbox` flag, at least when you download the snap.
See [diff](https://github.com/microsoft/vscode/compare/1.90.1...1.90.2) between old and fixed version and a later a [commit](https://github.com/microsoft/vscode/commit/4b19907cc8c2ce184326642f30ac606c3ebd2cb0) to change where `--no-sandbox` is defined

**Comment by luisriverag** [comment]
> Same here with ubuntu 24.04

Same

**Comment by Livia-Zaharia** [comment]
It is caused by messed up pemissions in AppImage, my workaround is https://github.com/getcursor/cursor/issues/1907

**Comment by danperks** [comment]
FYI closed a few duplicate issue reports, however a potential workaround not needing any AppArmour work was posted in #1907 if anyone with this issue wants to try that.

**Comment by antonkulaga** [comment]
Can developers just fix it or at least give a link to workaround? The problem is obvious: messed up AppImage release process when wrong permissions are given to part of the image. Having things broken for a huge part of users is not lovely. I am surprised that the issues persist for multiple months.

**Comment by lypanov** [comment]
FWIW easiest workaround is to run https://github.com/probonopd/go-appimage that'll auto create .desktop launchers that Just Work. I personally only needed the app armour config to ensure I could still start Cursor from the command line (and no longer do).

**Comment by helgster77** [comment]
@lypanov Would be great if you can provide detailed information about how you use go-appimage to make Cursor run. There are a few different tools in that repo.

**Comment by martinkrung** [comment]
I solved this with this work-around:

https://github.com/getcursor/cursor/issues/1683#issuecomment-2368366251

**Comment by nettum** [comment]
Lots of temporary fixes here now. Problem is that all of them are workarounds and needs work done by the end-user.   Some of the solutions also need to be re-done when cursor updates or user do manual update. Not a good first impression for new users that want to try the product.  
Hope the team will look into this as it's been broken for a very popular linux distro for some time now.  
I'm currently on the fence on cursor/upgrading plan. Need to get a feeling if linux is prioritized or not. 
Apart from this hiccup, very satisfied with cursor so far! :tada:

**Comment by jorcelinojunior** [comment]
To simplify and automate the process of managing app updates, I've created an interactive Bash script. This script enhances the user experience by allowing dynamic selection of specific tasks related to the Cursor app management.


<img src="https://github.com/user-attachments/assets/abb98f37-9291-4328-b335-69a81ebc7c90" alt="Options" width="600"/>


You can select:
- **Option (1)**: Search for and download the latest version if a newer one is available than what's currently on your machine.
- **Option (2)**: Re/install the latest version already downloaded to your machine and create a symbolic link to the executable with an icon.
- **Option (3)**: Configure AppArmor for the safe execution of the newly installed or updated version.

For more details, view the code and its explanation in my gist:

[cursor_appimage_manager.sh](https://gist.github.com/jorcelinojunior/131748e12a039c02148a6f66e1fb312c?permalink_comment_id=5206557#gistcomment-5206557)

I hope this can help someone.

Good luck!

**Comment by lypanov** [comment]
> @lypanov Would be great if you can provide detailed information about how you use go-appimage to make Cursor run. There are a few different tools in that repo.

Sorry I meant https://github.com/probonopd/go-appimage/blob/master/src/appimaged/README.md specifically.

**Comment by dangviethung096** [comment]
I searched this solution, i worked for me:
Ubuntu 24.04 implemented new restrictions for AppImage apps, which restricts the use of sandboxes.
run this command to reactive restriction:

`sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0`

https://github.com/electron/electron/issues/42510

**Comment by timoklein** [comment]
> Given all the hype about Cursor I'm kind of surprised, Ubuntu is not exactly a niche operating system.

Ran into the same issue and was like "Seriously?". Hope it gets fixed soon, really want to give Cursor a proper shot.

**Comment by 424778940z** [comment]
My company purchased multiple sets for Cursor sub., but your produce can't even run under a fresh install major Linux distro, this is unacceptable.

Cursor is not a game or office software, it's an IDE, you guys needs to realize there are Linux users, in fact a lot. The issue prisits for MONTHS,  telling us  won't fix try workarounds, is not a good impression for a commercial product.

Using an AI based product to save time and effort, but if we have to waste all that saved resources on figuring out how to get it running, then what's the point?

**Comment by ihatecsv** [comment]
Honestly it's not a great experience to download a broken appimage. Why not have a downloads page where you can write under the Linux download that you can't use Cursor on modern linux distros yet. That way we know ahead of time until you fix it.

The uni-download button is not great imo. What if I was looking for an exe but I'm running nix?

**Comment by antonkulaga** [comment]
People got 60mln of funding but cannot fix a bug that makes their system unusable and for which fixes were provided in the issues already ( @Livia-Zaharia gave a fix in one of her comments). Very disgraceful

**Comment by dalklein** [comment]
( @Livia-Zaharia gave a fix in one of her comments in September!).      Oh that is better than no-sandbox, thanks!

**Comment by lypanov** [comment]
If https://github.com/getcursor/cursor/issues/1907 is really the fix why was it closed so summarily?

I do not like to join it on +1 fests, but this is getting out of hand.

Your product still doesn't work on Linux.

**Comment by jurosh** [comment]
Based on how long this is open, it looks like team just doesn't care about Linux users. I wanted to try it too but ended up switching to MS Copilot after all the incompatibility hassle with the Cursor. And so far can recommend Copilot which can help with lot of small coding stuff.

**Comment by 424778940z** [comment]
> Based on how long this is open, it looks like team just doesn't care about Linux users. I wanted to try it too but ended up switching to MS Copilot after all the incompatibility hassle with the Cursor. And so far can recommend Copilot which can help with lot of small coding stuff.

well the thing is, it's just a fancy vscode fork that use few AI apis, we could just use those AI directly instead of using this

**Comment by lypanov** [comment]
> > Based on how long this is open, it looks like team just doesn't care about Linux users. I wanted to try it too but ended up switching to MS Copilot after all the incompatibility hassle with the Cursor. And so far can recommend Copilot which can help with lot of small coding stuff.
> 
> well the thing is, it's just a fancy vscode fork that use few AI apis, we could just use those AI directly instead of using this

Many of us our paid users that adore this product and just want to have a working install without workarounds.

To correct your misunderstanding, watch: https://www.youtube.com/watch?v=oFfVt3S51T4

**Comment by ihatecsv** [comment]
It's been 2 months since my last comment and no changes to support Linux users. Today I cancelled my Cursor subscription and started one with Windsurf Editor.

Value prop doesn't matter if you can't use the product.

**Comment by Dafaque** [comment]
> It's been 2 months since my last comment and no changes to support Linux users. Today I cancelled my Cursor subscription and started one with Windsurf Editor.
> 
> Value prop doesn't matter if you can't use the product.

My last month with cursor tho. Thanx for WE ref, gonna try taday.

**Comment by GauntletRunner2025** [comment]
Still occurring Jan 2025

**Comment by danperks** [comment]
Hey, this is actually an issue with Electron-based apps in general, and there is currently not a stable fix we can roll out, but we are keeping an eye on this to see if we can resolve this in the future!

See for reference:
https://github.com/electron/electron/issues/42510

**Comment by lypanov** [comment]
> Hey, this is actually an issue with Electron-based apps in general, and there is currently not a stable fix we can roll out, but we are keeping an eye on this to see if we can resolve this in the future!
> 
> See for reference: [electron/electron#42510](https://github.com/electron/electron/issues/42510)

IMHO https://github.com/electron/electron/issues/42510#issuecomment-2332919348 + https://github.com/gravitational/teleport/pull/43595 + should be a acceptable resolve. Many applications depend on Electron, and have applied such fixes.

**Comment by lypanov** [comment]
Spent the last hour reading into this and my recommendation to Cursor would be to provide a .deb package rather than AppImage specifically for Ubuntu users and use the fix described in https://github.com/electron/electron/issues/42510#issuecomment-2332919348 together with the AppArmor recommendation from here: https://github.com/gravitational/teleport/pull/43595.

If the investment this requires is too high, I'd ask Cursor to make it's position on such a contribution to this clear as no one wants to invest time in such PRs only to have them ignored.

**Comment by 424778940z** [comment]
> this is actually an issue with Electron-based apps in general

Then why vscode is running fine on my Linux PC? 
You could provide a deb/rpm package, or use provided workaround while packaging appimage.
But you chose to do nothing.

**Comment by GauntletRunner2025** [comment]
If its a known and expected error, maybe a warning or some kind of messaging around the download link could inform the user.  Or put it in a zip file along with a readme, or even a hint in the filename itself, better than nothing

**Comment by jorcelinojunior** [comment]
<p>I created a script that solves most of the common issues Linux users face with the <strong>Cursor .AppImage</strong> application, making everything easier! <br/> No more visiting the website to look for new versions or manually repeating tasks every time you want to update—this script handles all .AppImage-related updates and configurations for you.</p>

<h4>See <a href="https://github.com/jorcelinojunior/cursor-setup-wizard">Cursor Setup Wizard</a> in action below and see how it streamlines the entire process:</h4>

<p>
  <a href="https://github.com/jorcelinojunior/cursor-setup-wizard">
    <img src="https://raw.githubusercontent.com/jorcelinojunior/cursor-setup-wizard/main/images/cursor-setup-wizard-compressed.gif" alt="Cursor Setup Wizard in Action" width="920"/>
  </a>
</p>

<p>For full details and instructions, check out the repository's <strong><a href="https://github.com/jorcelinojunior/cursor-setup-wizard">README.md</a></strong>. Here's a quick overview:</p>

<p><strong>Highlights include:</strong></p>
<ul>
  <li><strong>Simplifies Cursor .AppImage management:</strong> Fetch updates, manage configurations, and resolve common issues effortlessly.</li>
  <li><strong>One command updates:</strong> Use the alias <code>cursor-setup</code> in your terminal to check for updates and install the latest <strong>.AppImage</strong> version anytime.</li>
  <li><strong>Optimized defaults:</strong> Works out of the box with pre-configured settings tailored for most users.</li>
  <li><strong>Customizable paths and themes:</strong> Easily modify paths, icons, and even colors to suit your preferences.</li>
  <li><strong>Multi-shell support:</strong> Automatically adds aliases for Bash and Zsh for quick access.</li>
  <li><strong>Detailed feedback:</strong> Logs every step of the process for transparency and easier troubleshooting.</li>
  <li><strong>Automatic version checks:</strong> Fetches the latest version online and compares its MD5 hash with your local version to ensure you're always up-to-date.</li>
  <li><strong>Comprehensive setup:</strong> Downloads the latest <strong>.AppImage</strong>, fetches the icon, creates desktop shortcuts, sets up AppArmor, and configures the "cursor" terminal command.</li>
</ul>

<p>Go to the <a href="https://github.com/jorcelinojunior/cursor-setup-wizard"><strong>cursor-setup-wizard repository</strong></a> and give it a try! 🚀</p>

<p>If you have any suggestions or run into issues, feel free to open an issue in the repository—I’ll gladly help!</p>

**Comment by amacfie-tc** [comment]
> I searched this solution, i worked for me: Ubuntu 24.04 implemented new restrictions for AppImage apps, which restricts the use of sandboxes. run this command to deactive restriction:
> 
> `sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0`
> 
> [electron/electron#42510](https://github.com/electron/electron/issues/42510)

does it need to be run after reboots?

**Comment by Enkumicahel** [comment]
> > I searched this solution, i worked for me: Ubuntu 24.04 implemented new restrictions for AppImage apps, which restricts the use of sandboxes. run this command to deactive restriction:
> > `sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0`
> > [electron/electron#42510](https://github.com/electron/electron/issues/42510)
> 
> does it need to be run after reboots?

This worked for me on Ubuntu 24.04. 

Thanks for sharing @amacfie-tc 🙏🏿

**Comment by antonkulaga** [comment]
>Thanks for sharing @amacfie-tc 🙏🏿

>This can be closed since it has nothing to do with Cursor.

I disagree. Ubuntu is major linux repo, and it is disgraceful that cursor team cannot properly package appimage for linux

**Comment by meawoppl** [comment]
Still an issue on:

```
meawoppl@XXXXX:~/$ sudo lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04.2 LTS
Release:	24.04
Codename:	noble
```

**Comment by vojavocni** [comment]
@antonkulaga Agreed, if this doesn't get resolved soon, I am moving away from cursor.

**Comment by Peter3412** [comment]
`--disable-accelerated-video-decode` works for me, same as vs-code

**Comment by inimitable** [comment]
I thought this must have been some weird one-off bug, but nope, apparently they're not packaging right for Ubuntu.

Okay, folks, if your business is an AI code completion editor that lets you be a 10xer, and you can't package your app for the most common distro... are you being a 10xer yet? :)

**Comment by yuchenli-sony** [comment]
still bugs can't fix?

**Comment by Dafaque** [comment]
Refound?

**Comment by drzraf** [comment]
If upstream devs' were to fix this bug, thousands of Linux developers would stop using `--no-sandbox` keeping Cursor 177MB opaque binary from doing:
- calling arbitrary process
- read/write arbitrary files
- upload / download arbitrary data over the wire

Preserving a buggy sandbox ensure sandbox deactivation so that the software _works as intended_ with extensive privileges. The same way a buggy release opens the door to awful `cursor-setup.sh`-like bash scripts popping up here and there.

Sounds more adequate to provide an **Apparmor** profile to somehow limit binary's accesses/calls.
(minimal base example here https://gist.github.com/drzraf/28fd981311e4cfb676c8f5281dfeca9f)
